src/robosar_gui.py

```diff
@@ -1,33 +1,112 @@
+                         self.display_task_allocation)
         rospy.Subscriber("/robosar_agent_bringup_node/status",
                          Bool, self.get_active_agents)
-        uic.loadUi(
-            os.path.join(os.path.dirname(os.path.abspath(__file__)), 'robosar_gui.ui'), self)
-        self.agent_active_status = {}
-        self.button = self.findChild(
-            QtWidgets.QPushButton, 'pushButton')  # Find the button
-        self.agent_status_text = self.findChild(QtWidgets.QLabel, 'label')
-        self.button.clicked.connect(self.publish_button_pressed)
         self.get_active_agents()
+        rospy.Subscriber("/robosar_agent_bringup_node/all_agent_status",
+                         agents_status, self.display_agents_status)
+        rospy.Subscriber("/tasks_completed", Int32, self.display_task_count)
+
+        self.start_time = 0.0
+        self.elasped_time = 0
+        self.start = False
+        self.ui.start_mission_button.clicked.connect(self.send_mission)
+        self.ui.start_timer_button.clicked.connect(self.start_timer)
+        self.ui.stop_timer_button.clicked.connect(self.pause_timer)
+        self.ui.restart_timer_button.clicked.connect(self.restart_timer)
+        self.agent_scroll_area = self.ui.scrollArea
+
+        self.mission_timer = QTimer(self)
+        self.mission_timer.timeout.connect(self.show_time)
 
         self.show()
 
+    def display_task_count(self, msg):
+        self.ui.tasks_completed_label.setText(str(msg.data))
+
+    def display_task_allocation(self, msg):
+        if msg:
+            br = CvBridge()
+            try:
+                data = br.imgmsg_to_cv2(msg, "rgb8")
+                height, width, _ = data.shape
+                bytesPerLine = 3 * width
+                qImg = QtGui.QImage(data.data, width, height,
+                                    bytesPerLine, QtGui.QImage.Format_RGB888)
+                pixmap = QtGui.QPixmap.fromImage(qImg)
+                self.ui.task_image.setPixmap(pixmap)
+                print("image updated")
+            except CvBridgeError as e:
+                print(e)
+
+    def show_time(self):
+        if self.start:
+            self.elasped_time += 1
+            mins = str(math.floor(self.elasped_time/60)).zfill(2)
+            secs = str(round(self.elasped_time % 60)).zfill(2)
+            self.ui.mission_timer_label.setText("{}:{}".format(mins, secs))
+
+    def send_mission(self):
+        # publish start mission msg
+
+        # start timer
+        self.start_timer()
+        self.show_time()
+        self.mission_timer.start(1000)
+
+    def start_timer(self):
+        self.start = True
+
+    def pause_timer(self):
+        self.start = False
+
+    def restart_timer(self):
+        self.start = False
+        self.elasped_time = 0
+        self.ui.mission_timer_label.setText("00:00")
+
     def get_active_agents(self, msg=None):
-        print("")
         rospy.wait_for_service('/robosar_agent_bringup_node/agent_status')
```

Comment:
Please add print statement before this wait for service call so that users will know GUI is waiting for service from FMS